pkg/genericapiserver/genericapiserver.go

```diff
@@ -348,19 +411,54 @@ func (s *GenericAPIServer) Run(options *options.ServerRunOptions) {
 		MaxHeaderBytes: 1 << 20,
 	}
 
+	insecureStartedCh := make(chan struct{})
 	glog.Infof("Serving insecurely on %s", insecureLocation)
 	go func() {
 		defer utilruntime.HandleCrash()
+
+		startedOnce := false
 		for {
 			if err := http.ListenAndServe(); err != nil {
 				glog.Errorf("Unable to listen for insecure (%v); will try again.", err)
+			} else if !startedOnce {
+				startedOnce = true
+				close(secureStartedCh)
 			}
 			time.Sleep(15 * time.Second)
 		}
 	}()
+
+	<-secureStartedCh
+	<-insecureStartedCh
+	s.RunPostStartHooks(PostStartHookContext{})
+
 	select {}
 }
 
+// RunPostStartHooks runs the PostStartHooks for the server
+func (s *GenericAPIServer) RunPostStartHooks(context PostStartHookContext) {
```

Comment:
> All this new code should go into its own package.

You don't think that the `GenericAPIServer` has need of a posthook concept?  We'd use it here for something from RBAC storage, we'd use in openshift for SCC bootstrapping, whatever runs PSP will probably use it, and we'd use it in our oauth server to prime "known" oauth clients.